### ISSUE TITLE
Rocker gesture to scroll through tab history

### ISSUE THREAD

**Issue by Reylak** [Robbendebiene/Gesturefy#99]
As a lot of people I think, I was forced to leave FireGestures behind, and I was happy to discover your addon was there!

Is it possible for rocker gestures to display a menu list in which we can scroll using the mouse wheel? If so, and similar to #91, I'd love to see my current tab's history and scroll through it in such a menu.

[I read from you](https://github.com/Robbendebiene/Gesturefy/issues/69#issuecomment-336711821) that you are limited by the API on this topic for now, so let's keep this issue as a reminder :)

**Comment by Robbendebiene** [comment]
> Is it possible for rocker gestures to display a menu list in which we can scroll using the mouse wheel? 

If I remember correctly this was named "popup gesture" in Firegestures, right?

> I read from you that you are limited by the API on this topic for now, 

Yeah, currently it's really hard to get the history of a tab (I would have to track all the tabs by myself)

**Comment by Reylak** [comment]
Yes, I believe it was popup gestures.

**Comment by Robbendebiene** [comment]
Because I can not use the normal contextmenu like Firegesture did I'll have to implement the popup menu by my self, wherefore you shoudln't expect this feature in the nearby future. But I like the idea and also made an implementation concept for this a while ago.

**Comment by nachoparker** [comment]
great! this would also be neat to have... of course, no pressure ;)

**Comment by NLZ** [comment]
Glad to hear that you are not rejecting this feature, as a FireGestures refugee I find this addon a great replacement aside from this. :)

**Comment by Robbendebiene** [comment]
I know a lot of you just want the little context menu back from Firegestures. But since I have to implement the interface/menu by myself, I want to encourage you to think of other possible ways to display the list of items. :)

To give some examples:
1. Open some kind of sidebar which contains the list of items, this sidebar could even cotain a little search at the top or bottom.
2. Only open a search field, which appears in the middle of the screen, which could work similar to firefox awesomebar (So you basicly would search through the item list)
3. Show a normal list, but divide it into columns if it would not fit into the screen length (this is obviously only usefull for a large item list)

I'm also open for any pros/cons of different ideas :)

**Comment by Robbendebiene** [comment]
Btw, what happened in Firegestures when the popup list was too long to be displayed on the screen? Was there a scrollbar in the popup menu?

**Comment by NLZ** [comment]
The point of it was that it was really quick to access, right under your mouse.
I'm not sure if the popup had scrollbar, but it was scrollable with mouse wheel.

**Comment by Robbendebiene** [comment]
@NLZ Or maybe it looked like the menu, which pops up when you click the arrow next to your tab bar? This is also scrollable but does not contain a scrollbar.

**Comment by Reylak** [comment]
@Robbendebiene the way FireGestures managed its popup actions, and which I loved, was that:
 * you pressed and held right mouse button (button was configurable I believe)
 * you scrolled at least once upward or downward
 * depending on the direction of the scroll (was configurable):
   * the history of the current tab was displayed
   * the list of currently open tabs was displayed
 * now (while still maintaining mouse button!) scrolling in whichever direction would scroll through the popup list
 * releasing the mouse button activated the highlighted entry (go back -- or go forward -- to page in history, or jump to open tab) and closes the popup list

About what happened when the list was too long, and went beyond the bottom screen edge, I believe it would simply slide the list upward when you reached the last visible entry. I'd even guess that wasn't managed by FireGestures itself, but rather by Firefox / its API / the OS's API. At any rate, no scroll bar was displayed. This behaved just as when you open a very long contextual menu in your OS, that sticks out of the screen.

Oh, and we were wrong: FireGestures actually called this "wheel gestures", although the associated action would mention "[Popup]" in its name, such as "[Popup] Back / Forward history" (I just checked [the website](http://www.xuldev.org/firegestures/) and its screenshots).

**Comment by nachoparker** [comment]
well, you could do it with 'wheel gestures' (hold a button and do wheel), or with 'rock gestures' (left-right or right-left). The popup was exactly like clicking the 'tab list' button on the upper right corner: no scroll bar, but scrollable if there's too many entries there's an indicator at the top/bottom that you can scroll, just like any menu like bookmarks and so on

**Comment by Robbendebiene** [comment]
@Reylak Wheel gestures are already available (with the same limitations as mouse gestures)

I'll try to implement something similar like the context menu/tab list menu. Thanks for the information @Reylak @nachoparker

**Comment by Robbendebiene** [comment]
Currently I wont be able to implement popup gestures that show the [history of the current tab](https://bugzilla.mozilla.org/show_bug.cgi?id=1378651) or lists the users [search engines](https://bugzilla.mozilla.org/show_bug.cgi?id=1352598). What else can be displayed in the popup?

**Comment by nachoparker** [comment]
I liked to use: current open tabs, recently closed tabs

thanks!

**Comment by svenschn** [comment]
I heavily used the recently closed tabs in Firegestures

**Comment by cy7yz2rj** [comment]
Recently closed tabs and currently open tabs for me as well.

**Comment by nachoparker** [comment]
GREAT JOB!!!

I you did such a good job, now I can right click gesture on Linux, and also I have my beloved popups for recently closed tabs and current tabs.

Thank you

**Comment by Robbendebiene** [comment]
@RheingoldRiver Yes and since the Mozilla devs have set the priority to 5, I wouldn't expect this to be implemented very soon.

**Comment by Andei** [comment]
Still restrictions from Mozilla for current tab history popup?

**Comment by Robbendebiene** [comment]
@Andei Yes, still no API available (one related bugzilla ticket: https://bugzilla.mozilla.org/show_bug.cgi?id=1378651)

**Comment by Err0rFlynnn** [comment]
Is editing the search engine popup still off the table? Where does it get the current list of search engines? I don't really need it to look up my browser search engines, just the ability to add one manually, like YouTube.

**Comment by Robbendebiene** [comment]
@Err0rFlynnn The popup displays all of your Firefox search engines similar to the address bar. You can find your added search engines here: `about:preferences#search`. You can also easily add further search engines like youtube to Firefox ([more info on how to do this can be found here](https://support.mozilla.org/en-US/kb/add-or-remove-search-engine-firefox))

Providing a custom list of search engines inside Gesturefy would require an extra UI which would require a lot of work.
